hack/jenkins/test-history/gen_history

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Compiles a static HTML site containing the last day's worth of test results.
+# Pass the URL of Jenkins into $1
+
+set -o errexit
+set -o nounset
+
+readonly jenkins="$1"
+readonly datestr=$(date +"%Y-%m-%d")
+
+mkdir -p static
+curl -fsS https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/test-history/gen_json.py > gen_json.py
+curl -fsS https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/test-history/gen_html.py > gen_html.py
+curl -fsS https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/test-history/index_template.html > index_template.html
+curl -fsS https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/test-history/static/style.css > static/style.css
+curl -fsS https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/test-history/static/script.js > static/script.js
+chmod +x gen_json.py
+chmod +x gen_html.py
+
+# Create JSON report
+time ./gen_json.py "${jenkins}" kubernetes
+
+# Create static HTML report out of the JSON
+./gen_html.py > static/tests.html
```

Comment:
They serve somewhat different purposes. This view shows us how a particular test is doing in all soak jobs, for instance, whereas the testgrid shows us how tests are doing in one soak job.